lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -7,12 +7,14 @@ const experimentalImportMetaResolve = getOptionValue('--experimental-import-meta
  * Generate a function to be used as import.meta.resolve for a particular module.
  * @param {string} defaultParentURL The default base to use for resolution
  * @param {typeof import('./loader.js').ModuleLoader} loader Reference to the current module loader
+ * @param {bool} allowParentURL Whether to permit parentURL second argument for contextual resolution
  * @returns {(specifier: string, parentURL?: string) => string} Function to assign to import.meta.resolve
  */
-function createImportMetaResolve(defaultParentURL, loader) {
-  return function resolve(specifier, parentURL = defaultParentURL) {
+function createImportMetaResolve(defaultParentURL, loader, allowParentURL) {
+  return function resolve(specifier) {
     let url;
 
+    const parentURL = allowParentURL ? arguments[1] ?? defaultParentURL : defaultParentURL;
```

Comment:
Didn't try it, but what about something like this?

```suggestion
    const parentURL = (allowParentURL && arguments[1]) ?? defaultParentURL;
```

Reply:
I think `&&` like the proposal are easily misread (the original is better—although I would toss in a couple line-breaks to make it more grokkable)

```suggestion
    const parentURL = allowParentURL ?
      arguments[1] ?? defaultParentURL :
      defaultParentURL;
```